source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,724 @@
 
-Cloud native setup automation
-*****************************
+#. Start the Docker service and enable it to start automatically at boot:
 
-Optional: Clone the `cloud-native-setup`_ repository on your system and follow the instructions. This repository includes helper scripts to automate configuration.
+   .. code::
 
-Package configuration customization (optional)
-**********************************************
+      sudo systemctl enable --now docker.service
 
-|CL| is a stateless system that looks for user-defined package configuration files in the :file:`/etc/<package-name>` directory to be used as default. If user-defined files are not found, |CL| uses the distribution-provided configuration files for each package.
+#. Configure kubelet to use the |CL| directory for cni-plugins and reload the
+   service.
 
-If you customize any of the default package configuration files, you **must** store the customized files in the :file:`/etc/` directory. If you edit any of the distribution-provided default files, your changes will be lost in the next system update as the default files will be overwritten with the updated files.
+   .. code-block:: bash
 
-For example, to customize CRI-O configuration in your system, run the following commands:
+      sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
 
-.. code-block:: bash
+      cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/0-cni.conf
+      [Service]                                                 
+      Environment="KUBELET_EXTRA_ARGS=--cni-bin-dir=/usr/libexec/cni"
+      EOF
 
-   sudo mkdir /etc/crio
-   sudo cp /usr/share/defaults/crio/crio.conf /etc/crio/
-   sudo $EDITOR /etc/crio/crio.conf
+   .. code:: bash
 
-Learn more about :ref:`stateless` in |CL|.
+      sudo systemctl daemon-reload
+
+
+#. Run the kubeadm command to initialize the master node:
+
+   .. important:: 
+
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. 
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is to use *flannel* as the pod networking. See `Choose a pod
+      network add-on`_ for more information.
+
+   .. code:: bash
 
-Proxy configuration (optional)
-******************************
+      sudo kubeadm init \
+      --pod-network-cidr 10.244.0.0/16 
 
-If you use a proxy server, you must set your proxy environment variables and create an appropriate proxy configuration file for both CRI-O services. Consult your IT department if you are behind a corporate proxy for the appropriate values. Ensure that your local IP is **explicitly included** in the environment variable *NO_PROXY*. (Setting *localhost* is not enough.)
 
-If you have already set your proxy environment variables, run the following commands as a shell script to configure all of these services in one step:
+#. (Optional) Docker can be set to use the kata-runtime as the default Docker
```

Comment:
This is a bad idea and is not recommended.  Using Kata with Docker with Kubernetes is not a recommended setup.  Please remove any Kata references.  Once you fix a runtime in Docker you can't change it and some pods need extra privileges.  This is why CRIO or containerd are the recommended setup for using Kata.

Reply:
I've removed this step.